models/PNNStack.py

```diff
@@ -112,11 +140,74 @@ def loss(self, pred, value):
         return F.l1_loss(pred, value)
 
     def loss_rmse(self, pred, value):
+
+        if self.inllloss == 1:
+            return self.loss_NLL(pred, value)
+        elif self.ihpwloss == 1:
+            return self.loss_hpweighted(pred, value)
+
         pred_shape = pred.shape
         value_shape = value.shape
         if pred_shape != value_shape:
             value = torch.reshape(value, pred_shape)
-        return torch.sqrt(F.mse_loss(pred, value))
+        return torch.sqrt(F.mse_loss(pred, value)), [], []
+
+    def loss_NLL(self, pred, value):  # negative log likelihood loss
```

Comment:
As we discussed, eventually this should be in the base class and used with any model (commenting to make sure there aren't any issues preventing that)